### ISSUE TITLE
Summary of todos about distributed tracing

### ISSUE THREAD

**Issue by drmingdrmer** [datafuselabs/databend#1227]
- [ ] Sometimes the spans messes up with the parent-child relationship. Maybe caused by abnormal thread switching? E.g. I found the parent span of `get_member_ship()`, which is a raft sub routine, somehow becomes the `SledTree:insert_values()`, which can not be possible.
- [ ] Chaining the span when a task was sent through channel. E.g., `async-raft` sends the `client_write` request through a channel to `LeaderState`, which is in another task. This results in a blank span lasting for several millisecond.
- [x] #1228
- [x] Use env var FUSE_JAEGER to specify a dedicate jaeger server or provide some way else.

**Comment by mapleFU** [comment]
I'm interested in this issue, if I want to encapsulate span in task context, what should I do?

**Comment by drmingdrmer** [comment]
> I'm interested in this issue, if I want to encapsulate span in task context, what should I do?

Generally speaking, a spawned task should be instrumented with a span. 
The way to do it is something like:
```rust
        let handle = tokio::spawn(
            async move { ...}.instrument(tracing::debug_span!("spawn")),
        );
```

A sample impl of this would be:
https://github.com/datafuse-extras/async-raft/blob/1492335798d9725c5d200277248e934a4c9247b9/async-raft/src/core/append_entries.rs#L251-L261

I've added several of cross task spans in to the **store** crate, in the latest developing branch, but not merged yet.
Because a recent bug I've tried to fix requires more detailed tracing info.

The second part is that when a **job** is transferred through channel, e.g., sending a log entry from the thread received it to the thread responsible to deal with it.

This part requires more thoughtful design. It is not that intuitive what should be done.
E.g., a span about sending a log entry to remote could belong to the ancestor span in which the log is emitted, or could also belong to the span of replication thread.
